pegasus/test/test_section_api_helpers.rb

```diff
@@ -498,6 +507,35 @@ class SectionApiHelperTest < SequelTestCase
           assert_nil row[:script_id]
         end
       end
+
+      it 'replaces a course assignment with a script assignment' do
+        Dashboard.db.transaction(rollback: :always) do
+          course_id = FakeDashboard::COURSES[0][:id]
+          script_id = FakeDashboard::SCRIPTS[0][:id]
+
+          params = {
+            user: {id: 15, user_type: 'teacher'},
+            course_id: course_id
+          }
+          section_id = DashboardSection.create(params)
+          row = Dashboard.db[:sections].where(id: section_id).first
+          assert_equal course_id, row[:course_id]
```

Comment:
Re: your aside - I have exactly that on line 523, or am I misunderstanding your comment?

This test is meant to be about the `update_if_owner` call. The asserts here are to ensure that our initial setup is in the right state before we make that call.